etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java

```diff
@@ -220,18 +222,27 @@ DomainResponse handleResults(DomainRequest request) {
 
     DomainResponse handleMetadata(DomainRequest request) {
         try {
-            String metadataId = request.getPathParams().get("id");
+            String receivedSubmissionId = request.getPathParams().get("id");
             Optional<PartnerMetadata> metadata =
-                    partnerMetadataOrchestrator.getMetadata(metadataId);
+                    partnerMetadataOrchestrator.getMetadata(receivedSubmissionId);
 
             if (metadata.isEmpty()) {
                 return domainResponseHelper.constructErrorResponse(
-                        404, "Metadata not found for ID: " + metadataId);
+                        404, "Metadata not found for ID: " + receivedSubmissionId);
             }
 
+            Set<String> messageIdsToLink =
+                    partnerMetadataOrchestrator.findMessagesIdsToLink(receivedSubmissionId);
+            Set<String> relevantMessageIdsToLinkInMetadata =
+                    messageIdsToLink.stream()
+                            .filter(s -> !s.equals(receivedSubmissionId))
+                            .collect(Collectors.toSet());
```

Comment:
Looks like you are just trying to remove the `receivedSubmissionId` from the set of IDs?  You can just do...

`messageIdsToLink.remove(receivedSubmissionId)`

Reply:
Updated